### ISSUE TITLE
Search engine optimization

### ISSUE THREAD

**Issue by Lcchy** [Mosc/Glider#176]
Hi there, first of all, thanks a lot for the nice app!

While comparing the search engine of the app with the official Hacker News search, I found that it was giving less accurate results (e.g. on "Gpt4-Turbo"). Additionally, it does not seem to be possible to search with "All Time" as a timespan.

I therefore wanted to ask, if Glider is using the official Hacker News search? And if not if there's a specific reason for it.

Thanks!

**Comment by Mosc** [comment]
I'm assuming that when you mention the "official Hacker News search", you're referring to the Hacker News search powered by Algolia at https://hn.algolia.com/. That would indeed be the same source that Glider is using for its search functionality.

I cannot reproduce the mentioned lower accuracy. I'm seeing basically the same results between this app and the web page. It would be helpful if you could provide some side-by-side results that show a quality discrepancy. The main difference between this app and the web page (with its default settings) is that Glider disables typo-tolerance because I found that that can result in some curious matches.

It _is_ possible to search without any time span, but it might not be immediately obvious if you haven't been exposed to the chip UI element much before. It's possible to deselect the currently selected date filter to enter the default "all time" mode.

**Comment by Lcchy** [comment]
Yes sorry for not being clear, I also corrected the issue title which was misleading.

Here are 2 screenshots comparing the search results:

![screenshot_2024-Jan-11th_13h11m52s](https://github.com/Mosc/Glider/assets/52164716/793425a3-ec4a-4dee-b40e-eae100018b9f)

![signal-2024-01-11-131223](https://github.com/Mosc/Glider/assets/52164716/9bfbe7dd-f96d-4f20-a6ab-2df3ad689991)


Maybe it is just a different ordering of results that is being used?

**Comment by Mosc** [comment]
I see. Initially you referenced "Gpt4-Turbo", which shows very similar results on Glider and Algolia, but you're right that "Gpt4 Turbo", without the dash, results in a lot more matches on the latter.

It does turn out that the typo-tolerance setting in Algolia is what makes up the difference. In this case the "correct" search term would be "GPT-4 Turbo". As mentioned I've had some weird results with typo-tolerance enabled, but there's an argument to be made that skipping the dash should be tolerated here. Let me meditate on it.

**Comment by Lcchy** [comment]
Ah yes I see. I wasn't sure in the beginning about what triggered the difference so I played around with it.
What you said makes sense, thanks!
